lib/test-observable.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Rx = require('rx');
+const compile = require('test262-compiler');
+const util = require('util');
+
+const scenariosForTest = require('./scenarios.js');
+
+module.exports = function(test262Dir, options = {}) {
+  const prelude = options.prelude || '';
+  const includesDir = options.includesDir;
+  const absolutePaths = (options.paths || [''])
+    .map(relativePath => path.join(test262Dir, relativePath));
+
+  return absolutePaths
+    .map(findFiles)
+    .reduce((accumulator, moreFiles) => accumulator.merge(moreFiles))
+    .map(pathToTestFile)
+    .map(test => compileFile(test, test262Dir, prelude, includesDir))
+    .flatMap(scenariosForTest);
+};
+
+function pathToTestFile(path) {
+  return { file: path, contents: fs.readFileSync(path, 'utf-8')};
+}
+
+function compileFile(test, test262Dir, prelude, includesDir) {
+  const endFrontmatterRe = /---\*\/\r?\n/g;
+  const match = endFrontmatterRe.exec(test.contents);
+  if (match) {
+    test.contents = test.contents.slice(0, endFrontmatterRe.lastIndex)
+                    + prelude
+                    + test.contents.slice(endFrontmatterRe.lastIndex);
+  } else {
+    test.contents = prelude + test.contents;
+  }
+  return compile(test, { test262Dir, includesDir });
+}
+
+function findFiles(targetPath, includeHidden) {
+  const files = new Rx.Subject();
+
+  fs.exists(targetPath, (exists) => {
+    if (!exists) {
+      files.onError(new Error('Path does not exist: "' + targetPath + '"'));
```

Comment:
Change

```js
'Path does not exist: "' + targetPath + '"'
```

to

```js
`Path does not exist: "${targetPath}"`
````

Reply:
Sure

Reply:
Agreed, but this was solved by an unrelated change